pkg/network/protocols/postgres/telemetry.go

```diff
@@ -97,3 +97,135 @@ func (t *Telemetry) Count(tx *ebpf.EbpfEvent, eventWrapper *EventWrapper) {
 func (t *Telemetry) Log() {
 	log.Debugf("postgres stats summary: %s", t.metricGroup.Summary())
 }
+
+// tlsAwareCounter has a plain counter and a counter for TLS.
+// It enables the use of a single metric that increments based on the encryption, avoiding the need for separate metrics for each use-case.
+type tlsAwareCounter struct {
+	counterPlain *libtelemetry.Counter
+	counterTLS   *libtelemetry.Counter
+}
+
+// NewTLSAwareCounter creates and returns a new instance of the struct
+func newTLSAwareCounter(metricGroup *libtelemetry.MetricGroup, metricName string, tags ...string) *tlsAwareCounter {
+	return &tlsAwareCounter{
+		counterPlain: metricGroup.NewCounter(metricName, append(tags, "encrypted:false")...),
+		counterTLS:   metricGroup.NewCounter(metricName, append(tags, "encrypted:true")...),
+	}
+}
+
+// Add adds the given delta to the counter based on the encryption.
+func (c *tlsAwareCounter) add(delta int64, isTLS bool) {
+	if isTLS {
+		c.counterTLS.Add(delta)
+		return
+	}
+	c.counterPlain.Add(delta)
+}
+
+// Get returns the counter value based on the encryption.
+func (c *tlsAwareCounter) get(isTLS bool) int64 {
+	if isTLS {
+		return c.counterTLS.Get()
+	}
+	return c.counterPlain.Get()
+}
```

Comment:
My original comment wasn't meant to support code duplication, moving `TLSAwareCounter` to a shared location is a good change, but should be done in a separate PR

Reply:
I see, I assumed I can do separate PR later. Do you want to make separate PR before this PR is done?

Reply:
yes please!

Reply:
done